script.js

```diff
@@ -1,4 +1,593 @@
+    } else if (currentPlayer > numberOfPlayers) {
+      myOutputValue = `The computer dealer will go next. Please click "submit" to continue.`;
+      currentGameStage = "computer turn";
+    }
+  }
+
+  //At computer turn stage.
+  //If computer initial hand score is < 17, draw card. Run Function 3b to draw 1 card, run Function 4b to calculate score.
+  //Keep doing above action till score >= 17.
+  //If computer hand score >=17, exit turn.
+  //Change game stage to compare score stage.
+  else if (currentGameStage == "computer turn") {
+    if (computerCurrentScore < 17) {
+      myOutputValue = `The computer dealer will draw cards as he has less than 17. <br> ${myComputerTurnImage} <br>Click "submit" to continue.`;
+      while (computerCurrentScore < 17) {
+        drawCard(shuffledDeck, computerCardArray);
+        computerCurrentScore = calculateScore(computerCardArray);
+      }
+    } else
+      myOutputValue = `Computer dealer turn has ended. Click "submit" to find out the winner!`;
+    currentGameStage = "compare scores";
+  }
+
+  //At compare score stage.
+  //Rules for player winning = player <= 21 and more than computer score, or computer score > 21 and player <=21
+  //Rules for player losing = player > 21 and computer <=21,  or player < computer score and computer score <=21
+  //Otherwise, tie.
+  //First see if computer > 21 or less than 21.
+  //If computer > 21, run loop to determine outcome of each player. Player can either win or tie. Assign coins according to bets made and outcome. Start creating output sentences for each player, with outcome and score.
+  //If computer <=21, run loop to determine outcome of each player. Player can win, lose or tie. Assign coins according to bets made and outcome. Start creating output sentences for each player, with outcome and score.
+  else if (currentGameStage == "compare scores") {
+    //Create output sentence for computer's cards in computer hand.
+    var computerSentence = outputResult(computerCardArray);
+    myOutputValue = "";
+    if (computerCurrentScore > 21) {
+      var z = 1;
+      while (z <= numberOfPlayers) {
+        if (allPlayerScoreArray[z - 1] <= 21) {
+          if (
+            allPlayerScoreArray[z - 1] == 21 &&
+            allPlayerCardArray[z - 1].length == 2
+          ) {
+            allPlayerCoins[z - 1] += Number(allPlayerBets[z - 1]) * 1.5;
+          } else {
+            allPlayerCoins[z - 1] += Number(allPlayerBets[z - 1]);
+          }
+          myOutputValue += `Player ${z} wins! <br> ${myWinImage} <br> Your score is ${
+            allPlayerScoreArray[z - 1]
+          } and you are closer to 21 than the dealer is! <br><br><br>  `;
+        } else if (allPlayerScoreArray[z - 1] > 21) {
+          allPlayerCoins[z - 1] += 0;
+          myOutputValue += `Player ${z} ties! <br> ${myTieImage} <br> Your score is ${
+            allPlayerScoreArray[z - 1]
+          }. <br><br><br>`;
+        }
+        z += 1;
```

Comment:
Blocks like this that are pretty long can probably be abstracted out as it's own function. as the game is pretty complicated, there's alot going on, abstracting these blocks out would help to declutter the main function